### PR TITLE
Implement Profile Picture: HTTP Post Call Handling

### LINKED ISSUE #7: A1.7 Profile Picture: HTTP Post Call Handling
﻿**Story:** ﻿As a User (Mentor, Mentee, or Educator), I want to be able to upload a professional profile picture of myself or something representing me.
﻿
﻿﻿**Details:**
﻿Allow for the retrieval of an image to be stored and for the previous image to be deleted within the database.
﻿
﻿Note: ﻿Allow for the removal of the image if no image is provided and the user specified removing the image, replace with default image.

﻿**Technical Information:**
- Django
- Handle Post Call
- Error Handling as a response is expected for improper Post Calls
- Back-end

### PR DESCRIPTION
**Pull request by JRSiwiecki**
Resolves #7 

### FEATURE
- When a user wishes to upload a profile picture, the POST request for doing so should include their accountID, and the profilePicture (or lack-thereof) they wish to upload.
- If an accountID is not supplied in the POST request, an error will be returned.
- If the accountID does not exist in the database or if the profile picture uploaded is not of the correct file type, an error will be returned.
- If a profilePicture was not supplied and only an accountID was supplied, then their existing profile picture will be "replaced" will None, at which point the profile picture is considered to be using the default profile picture in the meshapp/public folder.
- If a profilePicture is supplied then when it is uploaded it is renamed to a random file name along with the accountID appended to the end of the new file name, and is stored in the database. The picture is also accessible from the media/images folder located in the root folder, the file path of which would also be it's URL.
- To prevent orphaned profilePictures from existing in the media/images folder, I have added a cleanup library that removes these photos when a new (or null) profilePicture is uploaded.
- This endpoint is accessible from doing a POST request at /profiles/profilePicture.

Migrations were also re-created and re-made to be more up to date with current database models.

### CODE REVIEW

**Comment by joshgarde** [comment]
Hold off on merging; I wanna play with this from a sec perspective

File: mesh/urls.py [hunk 1]
```diff
@@ -46,3 +49,8 @@
     path('settings/', include(accountsettings_urls)),
     path('confirmation/', include(confirmation_urls)),
 ]
+
+if settings.DEBUG:
+    urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
+else:
+    urlpatterns += staticfiles_urlpatterns()
```
[the remark this answers is not in the document]
Comment (JRSiwiecki) [reply mesh/urls.py:56]: This is what allows the media files in the media/image folder to be grabbed directly by a URL. Without it, these urls are not accessible.

File: mesh/settings.py [hunk 2]
```diff
@@ -12,9 +12,6 @@
 
 from pathlib import Path
 import os
-from dotenv import load_dotenv
-
-load_dotenv()
```
[the remark this answers is not in the document]
Comment (JRSiwiecki) [reply mesh/settings.py:17]: These lines are redunant as if the backend is set-up correctly as per the instructions using pipenv.

File: mesh/profiles/views.py [hunk 3]
```diff
@@ -44,3 +42,73 @@ def profile_picture(request):
...
+            return JsonResponse(response, status=400)
+
+        # If user doesn't upload profile picture, then their profile picture will remain null.
+        if "profilePicture" not in file_data:
+            profile = Profile.objects.get(accountID = accountID)
+            profile.image = None
+            profile.save()
+
+            response["status"] = "Success"
+            response["message"] = "Profile picture not specified, using default null value."
```
[the remark this answers is not in the document]
Comment (JRSiwiecki) [reply mesh/profiles/views.py:87]: Developer presented, front-end should be dealing with how they display errors (or success message) in those cases, I just like having the extra detail for debugging purposes.

File: mesh/profiles/views.py [hunk 4]
```diff
@@ -44,3 +42,73 @@ def profile_picture(request):
...
                 response.update({"message": "An account does not exist with this account ID."})
                 return JsonResponse(response)
+            
+        
+    """
+        Handles HTTP POST request
+        for profile pictures, allowing users to upload,
+        replace, and delete their profile pictures.
+    """
+    if request.method == "POST":
```
[the remark this answers is not in the document]
Comment (JRSiwiecki) [reply mesh/profiles/views.py:52]: GET requests are already handled in the same function, as it was handled in #267 Profile Picture HTTP GET call handling. DELETE and PATCH are both planned as a separate issue for when we re-work the project to use Django REST framework.

[the remark this answers is not in the document]
Comment (joshgarde) [reply mesh/profiles/views.py:52]: Right, but while it's not implemented, you should be returning a HTTP 405 for those methods to indicate they're not available. Currently the server returns a HTTP 500 for anything besides a GET/POST

**Pull request closed without merging**